src/providers/42.ts

```diff
@@ -0,0 +1,71 @@
+import { OAuth2Client } from "oslo/oauth2";
+
+import type { OAuth2Provider } from "../index.js";
+
+export class FortyTwo implements OAuth2Provider {
+	private client: OAuth2Client;
+	private clientSecret: string;
+
+	constructor(
+		clientId: string,
+		clientSecret: string,
+		options?: {
+			redirectURI?: string;
```

Comment:
Is `redirectURI` required or not (the docs you wrote seems to indicate that it is)?